### DISCUSSION TITLE
How to stop animation when button is pressed

### DISCUSSION THREAD

**Discussion by qasymcorp** [LanKuDot/game_modules#36] [Q&A] [answered]
Hello!
Thank you for creating such a wonderful asset!

I have a question regarding the scrolling in ScrollingList.

I have 5 buttons which I am scrolling through linearly (horizontally), and I want to make it such a way when button is pressed it is centered and gets elevated slightly, that it is to show user that the button is clicked.
I made it that way using coroutines (button callback function starts this coroutine):

<i>This script is attached to every listBox element</i>
```cs
public IEnumerator JumpAnimation(float units = 7f)
    {
        isAnimationRunning = true;

        // Can't click other buttons while animation proceeds
        BlockInteractionWithButtons();

        // Wait so that button is centered
        yield return new WaitForSeconds(0.4f); // <1>

        // newPosition is above it's previous position
        Vector3 newPosition = new(rectTransform.position.x, rectTransform.position.y + units, rectTransform.position.z); // <2>

        // Smoothing animation
        while (newPosition != rectTransform.position)
        {
            rectTransform.position = Vector3.MoveTowards(
                rectTransform.position, 
                newPosition,
                animationSpeed);

            yield return null;
        }

        // As soon as animation finishes, user can click other buttons
        EnableInteractionWithButtons();

        isAnimationRunning = false;
    }
```

I want to make sure that user can't scroll through the list while animation is active, if I don't do so, user can create a mess by clicking a button and then quickly scrolling (this results in button moving towards the wrong place diagonally, without going back to normal)

At `<1>` I have a question; how do I know that the centering is finished?
At `<2>` How do I block scrolling so that the button does not move towards newPosition if user scrolls immediately after pressing?

**Comment by LanKuDot** [comment, marked as answer]
Hi, For 1. when the movement of the list is ended, it will send `OnMovementEnd` event.
For 2. add a flag (i.e. `_blockInput`) to the `CircularScrollingList.cs`, and create a function to toggle this flag.
Then check the flag in the following functions: `MoveOneUnitUp`, `MoveOneUnitDown`, `OnBeginDrag`, `OnDrag`, `OnEndDrag`, and `OnScroll`. It will be like:

```csharp
public void MoveOneUnitUp()
{
    if (_hasNoContent || _blockInput)
        return;

    ...
}
```

**Reply by qasymcorp** [reply]
Thank you for your response!!
